server/src/main/java/org/elasticsearch/action/search/SearchShardIterator.java

```diff
@@ -59,7 +60,9 @@ public SearchShardIterator(
         List<String> targetNodeIds,
         OriginalIndices originalIndices,
         ShardSearchContextId searchContextId,
-        TimeValue searchContextKeepAlive
+        TimeValue searchContextKeepAlive,
+        boolean prefiltered,
+        boolean skip
```

Comment:
A minor nice-to-have, but it would be good to move the javadoc from the other constructor down to this one and document the new parameters.

I had to go down to the "getter" methods for `skip` and `prefiltered` to understand what they represented.

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/95894/commits/bc808bb4100609b4e5d45911e31d7e20819f0503